UnityProject/Assets/Scripts/Objects/Botany/SeedExtractor.cs

```diff
@@ -139,39 +154,55 @@ public void DispenseSeedPacket(SeedPacket seedPacket)
+			if (interaction.HandObject.TryGetComponent<SeedPacket>(out var packet))
 			{
-				var grownFood = interaction.HandObject.GetComponent<GrownFood>();
-				var foodAtributes = grownFood.GetComponentInParent<ItemAttributesV2>();
-
-				if (!Inventory.ServerTransfer(interaction.HandSlot, storage.GetBestSlotFor(interaction.HandObject)))
-				{
-					Chat.AddActionMsgToChat(interaction.Performer,
-						$"You try and place the {foodAtributes.ArticleName} into the seed extractor but it is full!",
-						$"{interaction.Performer.name} tries to place the {foodAtributes.ArticleName} into the seed extractor but it is full!");
-					return;
+				AddSeedPacketToStorage(packet, interaction);
+				return;
 			}
+			var grownFood = interaction.HandObject.GetComponent<GrownFood>();
+			var foodAtributes = grownFood.GetComponentInParent<ItemAttributesV2>();
 
+			if (!Inventory.ServerTransfer(interaction.HandSlot, storage.GetBestSlotFor(interaction.HandObject)))
+			{
 				Chat.AddActionMsgToChat(interaction.Performer,
+					$"You try and place the {foodAtributes.ArticleName} into the seed extractor but it is full!",
+					$"{interaction.Performer.name} tries to place the {foodAtributes.ArticleName} into the seed extractor but it is full!");
+				return;
+			}
+
+			Chat.AddActionMsgToChat(interaction.Performer,
 				$"You place the {foodAtributes.ArticleName} into the seed extractor",
 				$"{interaction.Performer.name} places the {foodAtributes.name} into the seed extractor");
-				if (foodToBeProcessed.Count == 0 && currentState != PowerState.Off)
-				{
-					Chat.AddLocalMsgToChat("The seed extractor begins processing", gameObject);
-				}
-				foodToBeProcessed.Enqueue(grownFood);
-				return;
+			if (foodToBeProcessed.Count == 0 && currentState != PowerState.Off)
+			{
+				Chat.AddLocalMsgToChat("The seed extractor begins processing", gameObject);
 			}
-			// If no interaction happens
 			networkTab.ServerPerformInteraction(interaction);
+			foodToBeProcessed.Enqueue(grownFood);
+		}
+
+		private void AddSeedPacketToStorage(SeedPacket packet, HandApply interaction)
+		{
+			if (Inventory.ServerTransfer(interaction.HandSlot, storage.GetBestSlotFor(interaction.HandObject)))
+			{
+				seedPackets.Add(packet);
+				UpdateEvent.Invoke();
+				Chat.AddActionMsgToChat(interaction.Performer,
+					$"You place the {packet.gameObject.ExpensiveName()} into the seed extractor.",
+					$"{interaction.Performer.name} places the {packet.gameObject.ExpensiveName()} into the seed extractor.");
+				return;
+			}
+			Chat.AddActionMsgToChat(interaction.Performer,
+				$"You try and place the {packet.gameObject.ExpensiveName()} into the seed extractor but it is full!",
+				$"{interaction.Performer.name} tries to place the {packet.gameObject.ExpensiveName()} into the seed extractor but it is full!");
 		}
 
 		public bool WillInteract(HandApply interaction, NetworkSide side)
 		{
+			if (interaction.HandObject == null) return false;
```

Comment:
Do you not need to click the seed extractor with an empty hand to use it?

Reply:
I'll be honest, i have no clue how the extractor UI shows up when they should return false and gillie's comment confused me for a while after i started working on his review

Reply:
Before approving this I want to understand how and why this works incase this shows any troubles in the future.

Reply:
when I reviewed I mentioned how you had an ``if (interaction.HandObject == null)`` in ``ServerPerformInteraction`` and how it won't run because of the return you have highlited here. What confused you?

Reply:
I mean that im confused how this works regardless of any cases

Reply:
![image](https://user-images.githubusercontent.com/34368774/139495692-269438bf-0f87-46cb-91e4-776366b62de6.png)
Found the source of our problems, the UI works because there is another component that spawns it